### DISCUSSION TITLE
Simplify partial model queries

### DISCUSSION THREAD

**Discussion by Goodjooy** [SeaQL/sea-orm#1595] [Ideas]
## Motivation
When I writing my project. I have a complex Table like following.
```rust
#[derive(Debug, Clone, PartialEq, Eq, DeriveEntityModel, SubModel)]
#[sea_orm(table_name = "...")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub platform: String,
    pub datasource: String,
    pub nickname: String,
    pub avatar: String,
    pub config: String,
    pub unique_id: Uuid,
    pub db_unique_key: String,
    /// field for soft delete
    pub delete_at: DateTime,
} 
```
In many situation, I do not need all of those columns. If I only need `id` and `avatar` and `unique_id` info, I need a type like following 
```rust
#[derive(Debug, FromQueryResult)]
struct ModelFoo{
    pub id: i32,
    pub avatar: String,
    pub unique_id: Uuid,
}
```
Then, when I want query a `ModelFoo`， I may write following code
```rust
    Entity::find()
        .filter(Condition::all())
        .select_only()
        .column(Column::Id)
        .column(Column::Avatar)
        .column(Column::UniqueId)
        .into_model::<ModelFoo>()
        .all(&db)
```
That is a bit duplicating. I have been define the columns what I want but I still need wiring all columns I needing when query database. I think provide derive macro and its traits can reduce duplicating in this case.

## Proposed Solutions

I had writen a simple derive macro try to solve it. Currently it not support the result of join
https://github.com/Enraged-Dun-Cookie-Development-Team/Ceobe-Canteen-Serve/tree/select-only/macros/select-only-model-derive

**Comment by tyt2y3** [comment]
Oh wow. That's a great solution. I agree with your approach.
Would you like to upstream your macro? I think it is worth merging into master for maintenance.
There might be some differences in coding convention from ours, but is very good otherwise.

Please open a PR and we can proceed.

**Reply by Goodjooy** [reply]
> Oh wow. That's a great solution. I agree with your approach. Would you like to upstream your macro? I think it is worth merging into master for maintenance. There might be some differences in coding convention from ours, but is very good otherwise.
> 
> Please open a PR and we can proceed.

I had open a PR for this
 #1597
It is not finish yet, waiting for your review

**Comment by billy1624** [comment]
Indeed a neat and useful macros!
